tmd/view/view.py

```diff
@@ -1115,10 +1113,166 @@ def polar_plot(pop, neurite_type="neurites", bins=20):
+        vert1 = seg[0][vert]
+        vert2 = seg[1][vert]
+
+        return ((horz1, vert1), (horz2, vert2))
+
+    segs = [_seg_2d(seg) for seg in tree.get_segments()]
+
+    # Definition of the linewidth according to diameter, if diameter is True.
+
+    linewidth = [d * 1.0 for d in tree.d]
+
+    # Plot the collection of lines.
+    collection = _LC(segs, color=treecolors, linewidth=linewidth, alpha=1.0)
+
+    ax.add_collection(collection)
+
+    ax.set_xlim(bounding_box[0][term_dict[plane[0]]], bounding_box[1][term_dict[plane[0]]])
+    ax.set_ylim(bounding_box[0][term_dict[plane[1]]], bounding_box[1][term_dict[plane[1]]])
+
+    kwargs["title"] = kwargs.get("title", "Tree structure")
+    kwargs["xlabel"] = kwargs.get("xlabel", plane[0])
+    kwargs["ylabel"] = kwargs.get("ylabel", plane[1])
+
+    white_space = _get_default("white_space", **kwargs)
+    kwargs["xlim"] = kwargs.get(
+        "xlim",
+        [
+            bounding_box[0][term_dict[plane[0]]] - white_space,
+            bounding_box[1][term_dict[plane[0]]] + white_space,
+        ],
+    )
+    kwargs["ylim"] = kwargs.get(
+        "ylim",
+        [
+            bounding_box[0][term_dict[plane[1]]] - white_space,
+            bounding_box[1][term_dict[plane[1]]] + white_space,
+        ],
+    )
+
+    return cm.plot_style(fig=fig, ax=ax, **kwargs)
+
+
+def tree_barcode_colors(tree, plane="xy", feature="path_distances", cmap=cm.jet_map):
+    """Generate a 2d pic of the tree, each branch has a unique color.
+
+    Generate a barcode using the corresponding colors.
+    """
+    # pylint: disable=redefined-outer-name, import-outside-toplevel, unused-argument
+    from tmd.Topology.methods import _filtration_function
+    from tmd.Topology.methods import tree_to_property_barcode as tp_barcode
+
+    # Extract ph and ph_graph
+    ph, ph_graph = tp_barcode(tree, filtration_function=_filtration_function(feature))
+    colors_random = [cmap(i / len(ph)) for i in np.arange(len(ph))]
+
+    fig, ax = tree_colors(tree, ph_graph, colors=colors_random, new_fig=True, subplot=(211))
+
+    _ = fig.add_subplot(212)
+    plot.barcode(ph, color=colors_random, new_fig=False, subplot=(212))
+
+    return fig, ax
+
+
+def tree_full_persistence_colors(tree, plane="xy", feature="path_distances", cmap=cm.jet_map):
+    """Generate a 2d pic of the tree, each branch has a unique color.
+
+    Generate a barcode using the corresponding colors.
+    """
+    # pylint: disable=redefined-outer-name, import-outside-toplevel, unused-argument
+    from tmd.Topology.methods import _filtration_function
+    from tmd.Topology.methods import tree_to_property_barcode as tp_barcode
+
+    # Extract ph and ph_graph
+    ph, ph_graph = tp_barcode(tree, filtration_function=_filtration_function(feature))
+    colors_random = [cmap(i / len(ph)) for i in np.arange(len(ph))]
+
+    fig, ax = tree_colors(tree, ph_graph, colors=colors_random, new_fig=True, subplot=(221))
+
+    _ = fig.add_subplot(222)
```

Comment:
I guess we don't have to keep this result?

Reply:
yeah no need, I originally used to change titles but now that is done internally so it can be `_`